src/wp-includes/pluggable.php

```diff
@@ -91,18 +94,29 @@ function get_userdata( $user_id ) {
 	 *
 	 * @since 2.8.0
 	 * @since 4.4.0 Added 'ID' as an alias of 'id' for the `$field` parameter.
+	 * @since 5.3.0	Return the global $current_user if it's the user being fetched.
 	 *
 	 * @param string     $field The field to retrieve the user with. id | ID | slug | email | login.
 	 * @param int|string $value A value for $field. A user ID, slug, email address, or login name.
 	 * @return WP_User|false WP_User object on success, false on failure.
 	 */
 	function get_user_by( $field, $value ) {
+		global $current_user;
+
 		$userdata = WP_User::get_data_by( $field, $value );
 
 		if ( ! $userdata ) {
 			return false;
 		}
 
+		if ( ! empty( $current_user )
+			&& $current_user instanceof WP_User
+			&& $current_user->ID == $userdata->ID ) {
```

Comment:
```suggestion
		if (
			! empty( $current_user ) &&
			$current_user instanceof WP_User &&
			$current_user->ID === $userdata->ID
		) {
```

Unsure if the preference is for `is_a()` over `instanceof`, I saw one or the other being changed in a coding standards ticket during 5.6.

Reply:
I think it is `instanceof`. Looking at core, I'm finding 5 usages of `is_of` outside of library and test code and substantially more occurrences of `instanceof`.